docs/src/docs/user-guide.adoc

```diff
@@ -0,0 +1,694 @@
+
+The sample below shows how @JsonbCreator annotation can be used on a custom constructor. @JsonbProperty annotation on the constructor parameter is required for proper parameter substitution. In this case a value of JSON property 'name' will be passed to the constructor.
+
+[source,java]
+----
+public class Person {
+    private String name;
+    private String profession;
+
+    @JsonbCreator
+    public Person(@JsonbProperty("name") String name) {
+        this.name = name;
+    }
+}
+----
+
+==== Date/Number Formats
+
+By default JSON-B uses ISO formats to serialize and deserialize date and number fields. Sometimes it's required to override these settings. It can be done using @JsonbDateFormat and @JsonbNumberFormat annotations on fields:
+
+[source,java]
+----
+public class Person {
+    public String name;
+
+    @JsonbDateFormat("dd.MM.yyyy")
+    private Date birthDate;
+
+    @JsonbNumberFormat(“#0.00")
+    public BigDecimal salary;
+}
+----
+
+or globally using withDateFormat method of JsonbConfig class:
+
+[source,java]
+----
+// Create custom configuration
+JsonbConfig config = new JsonbConfig()
+    .withDateFormat("dd.MM.yyyy", null);
+
+// Create Jsonb with custom configuration
+Jsonb jsonb = JsonbBuilder.create(config);
+
+...
+----
+
+
+==== Binary Encoding
+
+JSON-B supports mapping of binary data. By default it uses BYTE encoding, but it can be easily customized using BinaryDataStrategy class which provides support for the most common binary data encodings:
+
+- BYTE
+- BASE_64
+- BASE_64_URL
+
+The sample below shows a creation of Jsonb engine with BASE_64_URL encoding:
+
+[source,java]
+----
+// Create custom configuration
+JsonbConfig config = new JsonbConfig()
+    .withBinaryDataStrategy(BinaryDataStrategy.BASE_64);
+
+// Create Jsonb with custom configuration
+Jsonb jsonb = JsonbBuilder.create(config);
+
+...
+----
+
+
+==== Adapters
+
+In some cases it may be difficult to serialize/deserialize a class the desired way. Sometimes it's not possible to put custom annotations on the source code because you don't have access to it or custom annotations don't solve the problem. In this case adapters is another option to try.
+
+Adapter is a class implementing javax.json.bind.adapter.JsonbAdapter interface. It has a custom code to convert the “unmappable” type (Original) into another one (Adapted) that is serialized/deserialized the desired way. It's sililar how type adapters in JAXB work.
+
+Let's take a look at the sample. Imagine that you have a Customer object with all customer details. In one scenario you need to serialize the whole object, in another you need to provide only its id and name. The solution could be to to serialize it as it is for the first scenario and create an adapter to JsonObject which has only data required for the second scenario.
+
+[source,java]
+----
+public class Customer {
+    private int id;
+    private String name;
+    private String organization;
+    private String position;
+
+    ...
```

Comment:
lets change this ellipsis to `// getters and setters ...` since we directly reference the getters and setters in the next code example